framework/framework.go

```diff
@@ -93,11 +94,38 @@ func New(config Config) (*Framework, error) {
 		return nil, microerror.Maskf(invalidConfigError, "config.ResourceRouter must not be empty")
 	}
 
+	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
+		ctx = canceledcontext.NewContext(ctx, make(chan struct{}))
+
+		if config.InitCtxFunc != nil {
+			var err error
+			ctx, err = config.InitCtxFunc(ctx, obj)
+			if err != nil {
+				return nil, microerror.Maskf(err, "initializing context")
+			}
+		}
+
+		accessor, err := meta.Accessor(obj)
+		if err != nil {
+			config.Logger.Log("warning", fmt.Sprintf("cannot create accessor for object %#v", obj))
+		} else {
+			meta, ok := loggermeta.FromContext(ctx)
+			if !ok {
+				meta = loggermeta.New()
+			}
+			meta.KeyVals["object"] = accessor.GetSelfLink()
```

Comment:
This looks like: `/api/v1/namespaces/default/pods/azure-operator-local-3998656933-2rr40`. I tested that on minikube.

Reply:
I am not sure this helps with grepping for the right logs. What I suggested initially was quite comfortable. 

```
grep \"framework\": \"pod\"
```

This would now be way more typing and always different because of the namespace. I do not see this is an improvement.
```
grep \"framework\": \"/api/v1/namespaces/default/pods
```

Reply:
`grep /pods/`, or `grep /kvms/` works and it's easy to type. Does that work for you?

Reply:
Grepping that way gives you anything that has `pods` and `kvms` in the log line. That is not really safe IMO.

Reply:
No, it gives you anything that has `/pods/` and `/kvms/`, and it's quite  safe for grep I think.

Reply:
If it isn't enough in the future, we can easily parse this endpoint and replace with different key-values, so I'd just go with that.

Reply:
Then lets iterate.